Add vars to remove default main cluster on Debian

l00ptr opened the pull request:
Define two vars to always be able to remove the default cluster
on Debian.

We no longer depend depend of postgresql_data_dir var. That way we
can delete the default cluster even when we move the data dir.

daniel-wind committed:
create users before database

So that the owner option can be used on de database playbook

ehelms committed:
Add Fedora support

daniel-wind committed:
Merge branch 'master' into patch-1

lejmr committed:
Adds support for naming PostgreSQL clusters

Commit:
Specify supported debian releases

lejmr committed:
Adds clean-ups and some asserts

lejmr committed:
PGPG for RedHat

lejmr committed:
Adds pgdg for Debian/Ubuntu

lejmr committed:
Clean-up for CentOS6

lejmr committed:
Expand test set

mikedep333 committed:
Add Fedora 30 support

Commit:
Add support for debian 10

geerlingguy committed:
Remove unused tests.

l00ptr committed:
Add vars to remove default main cluster on Debian

Define two vars to always be able to remove the default cluster
on Debian.

We no longer depend depend of postgresql_data_dir var. That way we
can delete the default cluster even when we move the data dir.

l00ptr committed:
Add vars to remove default main cluster on Debian

Define two vars to always be able to remove the default cluster
on Debian.

We no longer depend depend of postgresql_data_dir var. That way we
can delete the default cluster even when we move the data dir.

lejmr committed:
Fixes configuration issues

lejmr commented:
Can you please extend your code about ubuntu? Thank you

l00ptr commented:
yes i will extend for ubuntu. I don't use it and don't know if the package init an instance. I will check that

l00ptr committed:
Set default value for __postgres_version

  If we don't define a  postgres_version we will use the default
  version packaged by the distribution. Otherwise we override the
  value of the __postgres_version var because it's used to define
  other default variables (related to conf and data path)

l00ptr committed:
Set default value for __postgres_version

  If we don't define a  postgres_version we will use the default
  version packaged by the distribution. Otherwise we override the
  value of the __postgres_version var because it's used to define
  other default variables (related to conf and data path)

l00ptr committed:
Merge branch 'master' into add_vars_for_main_default_cluster_path

l00ptr commented:
it looks like there is some issues with the latest version of ansible (2.8.x) and molecule. 

Travis CI fails some people talk about this error there: https://github.com/ansible/molecule/issues/1727 and there: https://github.com/philpep/testinfra/issues/401

Do you know how to fix this issue ? What if we try with another version of ansible ?

l00ptr commented:
@lejmr any idea about how to fix this ? What about using an older version of ansible or maybe we can wait for a new version of molecule. I don't know if there are any other solution.

lejmr committed:
Adds support for extra options to initdb

lejmr committed:
Merge branch 'initdb_extra_options'

lejmr committed:
Ansible step for a nice configuration updates of postgresql.conf

lejmr committed:
Merge branch 'postgresql_config'

lejmr commented:
If the problem is due to molecule we can specify certain version of Molecule. The same thing we can do for Ansible pip package. I, however, prefer not to enforce ansible version, as we dont want to be fixed to a legacy versions.. Better option is to use "legacy" version of molecule I believe.. 

Ideally, we should define what versions of Ansible we support, so we test for them..

lejmr committed:
Downgrade of Molecule

l00ptr commented:
Hello @lejmr ,
indeed i don't want to define the version of ansible. The problem is related to testinfra (You can find more information there : [testinfra - issue 401](https://github.com/philpep/testinfra/issues/401)

Could you tell me if you can run the tests on your computer ? and if this is the case, could you give me more information about the version of Ansible, Molecule and Testinfra you use ?

I would like to add some features (what do you think about handling replication with this role ? ) to your role but not before we get this pr closed.

lejmr commented:
Tests are passing
```
testinfra==1.16.0
ansible==2.7.9
molecule==2.19.0
```

lejmr committed:
Merge branch 'pgdg' into add_vars_for_main_default_cluster_path

lejmr reviewed (commented):
Just wonder how to take my modification back because yours is better..  Although, I still dont understand the motivation for having the path to main cluster?

l00ptr commented:
@lejmr to remove it on debian based system. If we don't add the default path we override it we can't longer remove it. 

With the geerlingguy original role if you override postgresql_data_dir (to change the data location), you can't remove the default (main) cluster.

So the idea is to always keep the path of the default cluster even when we redifine the data location :-)

l00ptr commented:
travis test are using the latest version of ansible (2.8) so i guess this is why our tests fail.

lejmr commented:
Thank you for the explanation.. makes sense, so I am ok with that now :-)

l00ptr commented:
I will rebase from your master and check if everything is ok :-)

l00ptr committed:
Merge branch 'add_vars_for_main_default_cluster_path' of github.com:l00ptr/ansible-role-postgresql into add_vars_for_main_default_cluster_path

l00ptr committed:
Use ansible 2.7.10 for travis

geerlingguy committed:
Merge pull request #78 from ATIX-AG/add_debian_10

Add support for debian 10

Carbenium committed:
Resolve ansible deprecation warning

geerlingguy committed:
Merge pull request #81 from Carbenium/patch-1

Resolve ansible deprecation warning

geerlingguy committed:
PR #81-follow-up: Fix formatting.

mikedep333 committed:
Remove support for EOL Fedora 27 & 28

Also properly list support for Fedora 30, and improve
formatting in a when condition.

This addresses comments in PR86:
https://github.com/geerlingguy/ansible-role-postgresql/pull/86

mikedep333 committed:
Add fedora 29 to the test suite

I am not adding Fedora 30 because there is no image for it, only:
docker.io/geerlingguy/docker-fedora29-ansible

mikedep333 committed:
Add Fedora 30 to the test suite

`podman search` wasn't seeing the image, but the image exists:
https://hub.docker.com/r/geerlingguy/docker-fedora30-ansible

mikedep333 committed:
Fedora: Use python2-psycopg2 or python3-psycopg2

"python-psycopg2" is replaced by python2-psycopg2, and the
verification task will fail.

EL8 will need this also

mikedep333 committed:
Fix CI on Fedora 30 containers

by forcing the use of the C.UTF-8 locale, which is all they have
installed by default

mikedep333 committed:
Fix CI on Fedora 30 containers again

By not causing the /usr/bin/python symlink to be installed
Otherwise, Ansible molecule converge would use python2, which
lacks python2-dnf.

l00ptr committed:
Add debian 10 vars

Commit:
Remove symlinks to Debian config

Now that Debian 10 has been released, we should not need them.

Commit:
Fix test for empty list

I turns out that the bool filter casts a non-empty list to false. The
proper test seems to compare its length with zero.

geerlingguy committed:
Merge pull request #86 from mikedep333/fedora-30

Add support for Fedora 30

geerlingguy committed:
Merge pull request #92 from ATIX-AG/fix_empty_list_test

Fix test for empty list

geerlingguy committed:
Merge pull request #91 from ATIX-AG/remove_deb_prerelease_symlinks

Remove symlinks to Debian config

geerlingguy committed:
Remove Fedora 29 test since it fails more than 50% of the time.

geerlingguy committed:
Merge pull request #69 from ATIX-AG/specify_debian_releases

Specify supported debian releases

geerlingguy committed:
PR #69 follow-up: Corrected final list.

geerlingguy committed:
Fixes #95: Remove support for Ubuntu 14.04 Trusty.

geerlingguy committed:
Merge pull request #31 from daniel-wind/patch-1

create users before database

Commit:
Remove conflict marker

Commit:
Merge branch 'include_geerlingguy_changes'

Commit:
Merge branch 'master' of github.com:l00ptr/ansible-role-postgresql

Commit:
Merge branch 'master' into add_vars_for_main_default_cluster_path

Commit:
Add pgdg flag to postgresql molecule test

Commit:
Fix typo True -> tru

Commit:
Remove python library to handle postgres.conf

Commit:
Add with_items options for configure task
